src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -558,38 +621,261 @@ static void RemoveOtherSupportsOnDifferentPlatforms(SmallDictionary<string, Plat
                     symbol = accessor;
                 }
             }
-            var operationName = symbol.ToDisplayString(GetLanguageSpecificFormat(operation));
 
-            foreach (var platformName in attributes.Keys)
+            var originalAttributes = platformSpecificMembers[symbol] ?? attributes;
+
+            if (operation is IObjectCreationOperation creation)
             {
-                var attribute = attributes[platformName];
+                symbol = creation.Constructor.ContainingType;
+            }
+
+            var operationName = symbol.ToDisplayString(GetLanguageSpecificFormat(operation));
 
-                if (attribute.SupportedSecond != null)
+            foreach (var attribute in originalAttributes.Values)
+            {
+                if (AllowList(attribute))
                 {
-                    ReportSupportedDiagnostic(operation, context, operationName, platformName, VersionToString(attribute.SupportedSecond));
+                    ReportSupportedDiagnostic(operation, context, operationName, attributes, csAttributes);
                 }
-                else if (attribute.SupportedFirst != null)
+                else
                 {
-                    ReportSupportedDiagnostic(operation, context, operationName, platformName, VersionToString(attribute.SupportedFirst));
+                    ReportUnsupportedDiagnostic(operation, context, operationName, attributes, csAttributes);
                 }
+                break;
+            }
+
+            static void ReportSupportedDiagnostic(IOperation operation, OperationBlockAnalysisContext context, string operationName,
+                 SmallDictionary<string, PlatformAttributes> attributes, SmallDictionary<string, PlatformAttributes>? callsiteAttributes)
+            {
+                var supportedRule = GetSupportedPlatforms(attributes, callsiteAttributes, out var platformNames);
+                var platforms = string.Join(MicrosoftNetCoreAnalyzersResources.CommaSeparator, platformNames);
```

Comment:
Why do we need this to be localized? I think it would be fine to just use the hard coded string `, ` here and below when joining platform names.

Reply:
It was added by a [comment](https://github.com/dotnet/roslyn-analyzers/pull/4373#discussion_r518489206): `Assuming you use String.Join to create the string that represents the list of platforms, the item separator string also needs to be localizable. In English, it would be ", ", but there will be some languages that use a different separator string.`

Reply:
Thanks. I think we might have few other comma separated `string.Join` operations in the repo that use the hard coded string `, ` to generate the diagnostic argument.

Reply:
Let's confirm with @JasonCard. Do you want the `, ` separator to be localizable? Some localization teams I've worked with in the past have and some have not. If you don't think we need it to be a resource string, we can hard-code it.

Reply:
Comma can be different for Asian languages, so I believe it should be localizable, but I'm not familiar with this resource string and how it will be used, so @cristianosuzuki77 (loc owner for SW/product side) can double confirm.

Reply:
> Please add a real example as a resx comment, so the intent of asking the comma to be translated is clear to the translators as well.

Example added as a resx comment, how it is looks:
https://github.com/dotnet/roslyn-analyzers/blob/104dc0fc9785e7ea405607fbc6ff0970a1226b19/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx#L1489-L1491
FYI we are omitted `'or', 'and'` conjunctions for simplicity